integrations/src/test/java/io/jenkins/plugins/casc/SystemCredentialsTest.java

```diff
@@ -73,14 +73,13 @@ public void configure_system_credentials() throws Exception {
         List<BasicSSHUserPrivateKey> sshPrivateKeys = CredentialsProvider.lookupCredentials(
                 BasicSSHUserPrivateKey.class, jenkins, ACL.SYSTEM, Collections.emptyList()
         );
-        assertThat(sshPrivateKeys, hasSize(2));
-        final BasicSSHUserPrivateKey ssh_with_passphrase = sshPrivateKeys.stream()
-                .filter(k -> k.getId().equals("ssh_with_passphrase_provided"))
-                .findFirst().orElseThrow(AssertionError::new);
+        assertThat(sshPrivateKeys, hasSize(1));
+
+        final BasicSSHUserPrivateKey ssh_with_passphrase = sshPrivateKeys.get(0);
         assertThat(ssh_with_passphrase.getPassphrase().getPlainText(), equalTo("ABCD"));
 
         final DirectEntryPrivateKeySource source = (DirectEntryPrivateKeySource) ssh_with_passphrase.getPrivateKeySource();
-        assertThat(source.getPrivateKey(), equalTo("s3cr3t"));
+        assertThat(source.getPrivateKey().getPlainText(), equalTo("s3cr3t"));
```

Comment:
Not a blocker by any means, but it could be useful to create a matcher for secrets so you'd be able to assert something like `assertThat(source.getPrivateKey(), hasPlainText("s3cr3t"));` which would read better. Most appropriate place to add it would be [here](https://github.com/jenkinsci/jenkins-test-harness/blob/master/src/main/java/org/jvnet/hudson/test/JenkinsMatchers.java).